spam/03_spam_data_slicing_tutorial.py

```diff
@@ -0,0 +1,492 @@
+
+# %% [markdown]
+# ### Featurize Data
+#
+# First, we'll featurize the data—as you saw in the introductory Spam tutorial, we can extract simple bag of words features and store them as numpy arrays.
+
+# %%
+import torch
+from sklearn.feature_extraction.text import CountVectorizer
+
+vectorizer = CountVectorizer(ngram_range=(1, 1))
+
+
+def df_to_torch_features(vectorizer, df, fit_train=False):
+    words = [row.text for i, row in df.iterrows()]
+
+    if fit_train:
+        feats = vectorizer.fit_transform(words)
+    else:
+        feats = vectorizer.transform(words)
+    X = feats.todense()
+    Y = df["label"].values
+    return X, Y
+
+
+# %%
+X_train, Y_train = df_to_torch_features(vectorizer, df_train, fit_train=True)
+X_valid, Y_valid = df_to_torch_features(vectorizer, df_valid, fit_train=False)
+X_test, Y_test = df_to_torch_features(vectorizer, df_test, fit_train=False)
+
+# %% [markdown]
+# ### Create DataLoaders
+#
+# Next, we'll use the extracted Tensors to initialize a [`DictDataLoader`](https://snorkel.readthedocs.io/en/redux/packages/_autosummary/classification/snorkel.classification.DictDataLoader.html) — as a quick recap, this is a Snorkel-specific class that inherits from the common PyTorch class and supports multiple data fields in the `X_dict` and labels in the `Y_dict`.
+#
+# In this task, we'd like to store the bag-of-words `bow_features` in our `X_dict`, and we have one set of labels (for now) correpsonding to the `spam_task`.
+
+# %%
+from snorkel.classification.data import DictDataset, DictDataLoader
+
+BATCH_SIZE = 32
+
+
+def create_dict_dataloader(X, Y, split, **kwargs):
+    """Create a DictDataLoader for bag-of-words features."""
+    ds = DictDataset(
+        name="spam_dataset",
+        split=split,
+        X_dict={"bow_features": torch.FloatTensor(X)},
+        Y_dict={"spam_task": torch.LongTensor(Y)},
+    )
+    return DictDataLoader(ds, **kwargs)
+
+
+dl_train = create_dict_dataloader(
+    X_train, Y_train, split="train", batch_size=BATCH_SIZE, shuffle=True
+)
+dl_valid = create_dict_dataloader(
+    X_valid, Y_valid, split="valid", batch_size=BATCH_SIZE, shuffle=False
+)
+dl_test = create_dict_dataloader(
+    X_test, Y_test, split="test", batch_size=BATCH_SIZE, shuffle=False
+)
+
+# %% [markdown]
+# We can inspect our datasets to confirm that they have the appropriate fields.
+
+# %%
+dl_valid.dataset
+
+# %% [markdown]
+# ### Define [`MultitaskClassifier`](https://snorkel.readthedocs.io/en/redux/packages/_autosummary/classification/snorkel.classification.MultitaskClassifier.html)
+#
+# We define a simple Multi-Layer Perceptron (MLP) architecture to learn from the `bow_features`.
+#
+# _Note: the following might feel like extra steps to define what is a very simple architecture (e.g. `sklearn`'s [`MLPClassifier`](https://scikit-learn.org/stable/modules/generated/sklearn.neural_network.MLPClassifier.html)), but this will lend us additional flexibility later in the pipeline!_
```

Comment:
These italics aren't working in the render

Reply:
hm, it works on my end 
![image](https://user-images.githubusercontent.com/5819303/62838412-88bc3a80-bc30-11e9-8596-dfd8f27b44f7.png)

Reply:
not working on my github render...